### PR TITLE
Vendor library go changes for enable gcp and vsphere ccms

### PR DESCRIPTION
**Pull request by lobziik**
Updated library-go dependency with changes from openshift/library-go#1221. Needed for proceeding with GCP and vSphere out of tree providers related work.

### CODE REVIEW

File: pkg/controller/template/render_test.go [hunk 1]
```diff
@@ -48,11 +45,26 @@ func TestCloudProvider(t *testing.T) {
...
+		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
+		res:         "external",
+	}, {
+		platform:    configv1.OpenStackPlatformType,
+		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
+		res:         "external",
+	}, {
+		platform:    configv1.OpenStackPlatformType,
+		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
+		res:         "external",
```
Comment (kikisdeliveryservice) [inline pkg/controller/template/render_test.go:60]: Is there a reason why these two OpenStack cases seem to be identical? Looks like an accidental dupe?

**Review by kikisdeliveryservice — COMMENTED** [review]
test q

File: pkg/controller/template/render_test.go (hunk 1 above)
Comment (lobziik) [reply pkg/controller/template/render_test.go]: Yup, accidental. Fixed.

File: go.mod [hunk 2]
```diff
@@ -35,9 +35,9 @@ require (
 	github.com/imdario/mergo v0.3.12
 	github.com/mattn/go-isatty v0.0.12 // indirect
 	github.com/opencontainers/go-digest v1.0.0
-	github.com/openshift/api v0.0.0-20210921141042-49b82c6d1ca9
+	github.com/openshift/api v0.0.0-20210924154557-a4f696157341
```
Comment (sinnykumari) [inline go.mod:38]: Also need to pin this version by updating openshift/api version in replace section https://github.com/openshift/machine-config-operator/blob/master/go.mod#L38

File: pkg/controller/template/render_test.go [hunk 3]
```diff
@@ -31,9 +31,6 @@ func TestCloudProvider(t *testing.T) {
 		featureGate *configv1.FeatureGate
 		res         string
 	}{{
-		platform: configv1.AWSPlatformType,
```
Comment (sinnykumari) [inline pkg/controller/template/render_test.go:34]: I don't fully understand. Are these no longer in use i.e. cloudProvider ID will be only set using featuregate?

Comment (lobziik) [reply pkg/controller/template/render_test.go:34]: Just moved this case to line 61.

File: go.mod (hunk 2 above)
Comment (lobziik) [reply go.mod:38]: Done

**Review by sinnykumari — APPROVED** [review]
LGTM. Will wait for Kirsten in case there are some final questions/concerns.

**Review by kikisdeliveryservice — APPROVED** [review]
seems good to go now =)

/lgtm

**Pull request merged**